src/NServiceBus.Core/Routing/DirectRoutingTable.cs

```diff
@@ -0,0 +1,68 @@
+namespace NServiceBus.Routing
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Linq;
+
+    /// <summary>
+    /// Manages the direct routing table.
+    /// </summary>
+    public class DirectRoutingTable
+    {
+        List<Func<Type, IEnumerable<DirectRoutingDestination>>> rules = new List<Func<Type, IEnumerable<DirectRoutingDestination>>>();
```

Comment:
I think we should use our own representation of `MessageType` internally to avoid coupling us to much to CLR types?

You can see this happening to us in the pass where we eg. are passing the assembly qualified type in the headers which make it harder to interop (platform coupling)

https://github.com/Particular/NServiceBus/issues/1610